go/obscuronode/header.go

```diff
@@ -0,0 +1,106 @@
+package obscuronode
+
+import (
+	"sync"
+
+	common2 "github.com/obscuronet/obscuro-playground/go/obscuronode/common"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// BlockHeader is holds the header info for the l1 blocks
+type BlockHeader struct {
+	ID     common.Hash
+	Parent common.Hash
+	Height uint
+}
+
+// RollupHeader is holds the header info for the l2 rollups
+type RollupHeader struct {
+	ID     common.Hash
+	Parent common.Hash
+	Height uint
+
+	Withdrawals []common2.Withdrawal
+}
+
+// NodeHeader allows to access the nodes current l1, l2 headers
+type NodeHeader struct {
```

Comment:
can we call this Db?
to be consistent with how it's called everywhere

Reply:
Storage==Database

What do you mean a db that has a db?
In my mind this is the db of the host node.
We'd stick more stuff in it as we need it, and eventually back it by  leveldb or whatever

Reply:
It's an encapsulation point of view. 
It makes sense to keep it simple - reverted to DB :)

What I meant was I'd prefer to have 
```
type Node struct {
    type Storage struct {
          db DB.interface
          currentHeader header.interface
          ...
          nodePublicKey  PubKey.interface
    }
}

And then have
Node.Storage.GetCurrentHeader()
Node.Storage.GetPubkey()
```

Over 
```
type Node struct {
    type DB struct {
          db DB.interface
          currentHeader header.interface
          ...
          nodePublicKey  PubKey.interface
    }
}

And then have
Node.DB.GetCurrentHeader()
Node.DB.GetPubkey()
```

Reply:
gotcha. basically the business logic should be in the "storage" layer, and the "db" should just store without any logic.
That is definitely what we'll have eventually when we introduce leveldb